Add CI instructions to README (#173)
returntocorp/bento: README.md

</p>

## Installation

> ```bash

Reviewer: nit: we've been using shell, but bash is probably the same 🤷‍♂

$ bento archive
```

> Bento really sings when you run it automatically in your editor or as a commit hook (`bento install-hook`).

Reviewer 2: we should just link the Emacs integration to test if people would ask about it!

> Bento really sings when you run it automatically in your editor or as a commit hook (`bento install-hook`).

Reviewer: ```suggestion
Bento is at its best when run automatically as a commit hook (i.e. `bento install-hook`) or as part of CI.
```

> ### CI setup

Reviewer: ```suggestion
### Running Bento in CI
```

If you like what you see, try adding Bento to your CI pipeline. 

If you use CircleCI, you can add the following job:

```
version: 2.1

jobs:

>     bentoRun:

Reviewer: ```suggestion
    bentoCheck:
```

    executor: circleci/python:3.7.4-stretch-node
    steps:
      - checkout
      - run:

>           name: "Install bento-cli"

Reviewer: ```suggestion
          name: "Install Bento"
```

>           command: pip install bento-cli && bento --version

Author: ```suggestion
          command: pip3 install bento-cli && bento --version
```

      - run:

>           name: "Running bento check"

Reviewer: ```suggestion
          name: "Run Bento check"
```

^ making it consistent with language in the `bento init` mocks and fixing verb

          command: bento --agree --email <YOUR_EMAIL> check
```

Otherwise, you can simply install and run Bento in CI with the following commands:

```

> pip install bento-cli && bento --version

Author: ```suggestion
pip3 install bento-cli && bento --version
```

bento --agree --email <YOUR_EMAIL> check
```

> `bento check` will exit with an exit code of `2` if it finds any issues, and with an exit code of `3` if it fails to run. Address the issues or unblock yourself by running `bento archive` to add them to your `.bento-whitelist` file and addressing them later. Then check in your whitelist file.

Reviewer: For the exit codes, could we add a section to the README that articulates exit codes and have this section link to it?

> Please open an issue if you need help setting up Bento with another CI provider.

Reviewer: Can we add a link to filing an issue here? 

Perhaps invite others to submit docs for other providers if they configure Bento?

$ bento archive
```

Bento is at its best when run automatically as a commit hook (i.e. `bento install-hook`) or as part of CI.

### Exit Codes
`bento check` may exit with the following exit codes: 
- `0`: Bento ran successfully and found no errors
- `2`: Bento ran successfully and found issues in your code
- `3`: Bento or one of its underlying tools failed to run

### Running Bento in CI

> If you like what you see, try adding Bento to your CI pipeline. 

Reviewer 3: I think you don’t need this line here. I think it’s fine to start this section with the instruction.

Reviewer: +1

Reviewer: +1

> If you use CircleCI, you can add the following job:

Reviewer 3: ```suggestion
If you use CircleCI, add the following job:
```

> ```

Reviewer: ```suggestion
```yaml
```

Not sure if this works, but would be nice to mark this `yaml` for coloration.

version: 2.1

jobs:
    bentoCheck:
    executor: circleci/python:3.7.4-stretch-node
    steps:
      - checkout
      - run:
          name: "Install Bento"
          command: pip3 install bento-cli && bento --version
      - run:
          name: "Run Bento check"
          command: bento --agree --email <YOUR_EMAIL> check
```

> 

Reviewer 3: Just FYI I have not verified these steps as I am not a CircleCI user.

Otherwise, you can simply install and run Bento in CI with the following commands:

> ```

Reviewer: ```suggestion
```shell
```

pip3 install bento-cli && bento --version
bento --agree --email <YOUR_EMAIL> check
```

> `bento check` will exit with a non-zero exit code if it finds issues in your code (see [Exit Codes](#exit-codes)). You can run `bento --agree --email <YOUR_EMAIL> check || true` if you'd like to prevent Bento from blocking your build. Otherwise, address the issues or unblock yourself by running `bento archive` to add them to your `.bento-whitelist` file to address later. Don't forget to check in your whitelist file.

Reviewer: ```suggestion
`bento check` will exit with a non-zero exit code if it finds issues in your code (see [Exit Codes](#exit-codes)). You can run `bento --agree --email <YOUR_EMAIL> check || true` if you'd like to prevent Bento from blocking your build. Otherwise, address the issues or unblock yourself by running `bento archive`.
```

Future proofing by removing discussion of file name here :)
